test/framework-ondata-mocha.js

```diff
@@ -0,0 +1,526 @@
+            var wrapper = { fRunUpdater: fRunUpdater };
+            var node = test.createComponent( 
+                factory({
+                          description: "Test Description",
+                          inPorts: {
+                              input: {
+                                  datatype: 'string',
+                                  description: "a string port",
+                                  addressable: true,
+                                  required: true
+                              }
+                          }
+                        }, 
+                        wrapper )
+            );
+
+            // Attach 2 sockets to the same port and send each some data
+            return new Promise( function(done, fail) { 
+
+                // Monitor output and error port to see resulting status
+                test.onOutPortData(node, 'output', done);
+                test.onOutPortData(node, 'error', fail);
+
+                var socket1 = noflo.internalSocket.createSocket();
+                node._component_under_test.inPorts['input'].attach( socket1 );
+
+                var socket2 = noflo.internalSocket.createSocket();
+                node._component_under_test.inPorts['input'].attach( socket2 );
+
+                socket1.send( inputEdge1 );
+                socket2.send( inputEdge2 );
+
+            }).then( function( done ) { 
+
+               // Verify we got the output state we expect
+               done.vnid.should.equal( '' ); 
+               done.data.should.equal( executedFRunUpdater ); 
+
+               // Cleanup 
+               test.detachAllInputSockets( node );
+            }, function( fail ) { 
+               assert.isNotOk( fail );
+            });
+        });
+
+
+        it( "should fail gracefully when fRunUpdater throws an exception.", function() {
+
+            var inData = "A bit of input data";
+            var executedFRunUpdater = "fRunUpdater failed!!!";
+
+            // Define the fRunUpdater that framework should invoke
+            var fRunUpdater = function() { 
+                throw new Error( executedFRunUpdater );
+            }
+
+            // Create a noflo component and get the node instance for it
+            var wrapper = { fRunUpdater: fRunUpdater };
+            var node = test.createComponent( 
+                factory({
+                          description: "Test Description",
+                          inPorts: {
+                              input: {
+                                  datatype: 'string',
+                                  description: "a string port",
+                                  required: true
+                              }
+                          }
+                        }, 
+                        wrapper )
+            );
+
+            var logBuffer = '';
+            // Hide expected console error message from test output
+            sinon.stub( console, 'error', function (message) {
```

Comment:
I have it here because the error messages we are causing will print to the console and be pretty confusing to someone who does not know these tests.  They will wonder if there is an error here or not.  Let me show you what that would look like before we remove it.  It's not pretty.